### PR TITLE
Feat/gh8 hub browser; fixes RHEAGROUP/DEH-MDSYSML#8

### PR DESCRIPTION
**Pull request by nathanatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/DEH-CommonJ/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/DEH-CommonJ/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
;fixes RHEAGROUP/DEH-MDSYSML#8

- [x] added a tree view browser for the Element Definition tree

### CODE REVIEW

File: src/main/java/Reactive/ObservableCollection.java [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+
+    @Override
+    public ListIterator<TValue> listIterator(int index)
+    {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public List<TValue> subList(int fromIndex, int toIndex)
```
Comment (alexatrhea) [inline src/main/java/Reactive/ObservableCollection.java:308]: Some docs missing in this file.

File: src/main/java/Utils/ImageLoader/ImageLoader.java [hunk 2]
```diff
@@ -42,20 +44,49 @@
...
+     * @param subfolder the sub folder where the requested image resides
+     * @param fileName the file name of the requested image
+     * @return a {@link ImageIcon}
+     */
+    public static ImageIcon GetIcon(String subfolder, String fileName)
+    {
+        return GetIcon(ImageLoader.class.getResource(RootFolder+subfolder+fileName), fileName);
+    }
+
+
```
Comment (alexatrhea) [inline src/main/java/Utils/ImageLoader/ImageLoader.java:80]: double space

File: src/main/java/ViewModels/Interfaces/IObjectBrowserViewModel.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package ViewModels.Interfaces;
+
+import org.netbeans.swing.outline.OutlineModel;
+
+import io.reactivex.Observable;
+
+public interface IObjectBrowserViewModel extends IViewModel
```
Comment (alexatrhea) [inline src/main/java/ViewModels/Interfaces/IObjectBrowserViewModel.java:30]: missing docs

File: src/main/java/ViewModels/ObjectBrowser/ElementDefinitionTree/Rows/Parameters/ParameterValueBaseRowViewModel.java [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+                        break;
+                    case CompoundParameterType:
+                        valueRow.SetName(((CompoundParameterType)parameterType).getComponent().get(index).getShortName());
+                        break;
+                    default:
+                        valueRow.SetName(this.GetThing().getParameterType().getName());
+                        break;
+                }
+                
+
```
Comment (alexatrhea) [inline src/main/java/ViewModels/ObjectBrowser/ElementDefinitionTree/Rows/Parameters/ParameterValueBaseRowViewModel.java:241]: extra line

**Review by alexatrhea — APPROVED** [review]
Some editorial comments

File: src/main/java/Views/HubLogin.java [hunk 5]
```diff
@@ -410,10 +410,11 @@ public void actionPerformed(ActionEvent e)
                 CloseDialog(false);
             }
         });
-        
+
```
Comment (arielleatrhea) [inline src/main/java/Views/HubLogin.java:413]: double line

**Review by arielleatrhea — APPROVED** [review]

**Review by alexatrhea — APPROVED** [review]

**Pull request merged**